libp2p/protocols/pubsub/gossipsub/behavior.nim

```diff
@@ -329,14 +383,15 @@ proc handleIWant*(
           invalidRequests.inc()
           if invalidRequests > 20:
             libp2p_gossipsub_received_iwants.inc(1, labelValues = ["skipped"])
-            return messages
+            return result
           continue
         let msg = g.mcache.get(mid).valueOr:
           libp2p_gossipsub_received_iwants.inc(1, labelValues = ["unknown"])
           continue
         libp2p_gossipsub_received_iwants.inc(1, labelValues = ["correct"])
-        messages.add(msg)
-  return messages
+        result.messages.add(msg)
+        result.ids.add(mid)
```

Comment:
It's probably better to define a `messages` and `ids` variables, instead of using `result`. See https://status-im.github.io/nim-style-guide/language.result.html